lib/pr_checker.js

```diff
@@ -169,9 +169,31 @@ class PRChecker {
       logger.warn('No full CI runs detected');
     }
 
+    let lastCI;
     for (const [type, ci] of ciMap) {
       const name = CI_TYPES.get(type).name;
       logger.info(`Last ${name} CI on ${ci.date}: ${ci.link}`);
+      if (!lastCI || lastCI.date > ci.date) {
+        lastCI = {
+          typeName: name,
+          date: ci.date
+        };
+      }
+    }
+
+    if (lastCI) {
+      let didLogHeader = false;
+      commits.forEach((c) => {
+        const commit = c.commit;
+        if (commit.committedDate > lastCI.date) {
+          if (!didLogHeader) {
+            status = false;
+            didLogHeader = true;
+            logger.warn(`Commits pushed after the last ${lastCI.typeName} CI run:`);
```

Comment:
Logging this line in every iteration would bring too much noise, maybe just filter out the new commits and do the logging according to the length of the array?

Reply:
This logs just once due to `didLogHeader`.

Reply:
@apapirovski Ah right, I knew there should be a reason I didn't raise this question before lol